cpu/samd5x/cpu.c

```diff
@@ -191,37 +183,40 @@ static void dfll_init(void)
     while (!OSCCTRL->STATUS.bit.DFLLRDY) {}
 }
 
-static void fdpll0_init(uint32_t f_cpu)
+static void fdpll_init(uint8_t idx, uint32_t f_cpu)
 {
+    /* Trigger assertion if not using FDPLL0 or FDPLL1 */
+    assert(idx == 0 || idx == 1);
+
     if (!USE_DPLL) {
-        OSCCTRL->Dpll[0].DPLLCTRLA.reg = 0;
+        OSCCTRL->Dpll[idx].DPLLCTRLA.reg = 0;
         return;
     }
 
     /* We source the DPLL from 32kHz GCLK1 */
-    const uint32_t LDR = ((f_cpu << 5) / 32768);
+    const uint32_t LDR = (f_cpu / 32768);
```

Comment:
Alternatively you could have used
```suggestion
    const uint32_t LDR = (f_cpu >> 10);   /* LDR = ((f_cpu << 5) / 32768);
```
Should be the same as `((f_cpu << 5) / 32768)`. This might me more reproducible together with the former code for `DPLLRATIO`.

Reply:
I'll give it a try.